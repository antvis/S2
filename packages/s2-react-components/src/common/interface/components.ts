export interface BaseComponentProps<T> {
  /**
   * 标题
   * @default "文字对齐"
   */
  title?: React.ReactNode;

  /**
   * 默认是否折叠
   * @default false
   */
  defaultCollapsed?: boolean;

  /**
   * 默认配置
   */
  defaultOptions?: Partial<T>;

  /**
   * 子节点
   */
  children?: React.ReactNode;
}
