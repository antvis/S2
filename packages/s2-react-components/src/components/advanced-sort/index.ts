export { AdvancedSort } from './advanced-sort';
export { CustomSort } from './custom-sort';

export type {
  AdvancedSortBaseProps,
  AdvancedSortProps,
  CustomSortProps,
  Dimension,
  RuleItem,
  RuleOption,
  RuleValue,
} from './interface';
