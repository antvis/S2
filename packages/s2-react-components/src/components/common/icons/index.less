@import '@antv/s2-shared/src/styles/variables.less';

.@{icon-cls-prefix} {
  color: inherit;
}
