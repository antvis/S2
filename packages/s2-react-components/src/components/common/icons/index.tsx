import { ReactComponent as CalendarIcon } from '@antv/s2-shared/src/icons/calendar-icon.svg';
import { ReactComponent as ColIcon } from '@antv/s2-shared/src/icons/col-icon.svg';
import { ReactComponent as LocationIcon } from '@antv/s2-shared/src/icons/location-icon.svg';
import { ReactComponent as RowIcon } from '@antv/s2-shared/src/icons/row-icon.svg';
import { ReactComponent as TextIcon } from '@antv/s2-shared/src/icons/text-icon.svg';
import { ReactComponent as ValueIcon } from '@antv/s2-shared/src/icons/value-icon.svg';
import './index.less';

export { CalendarIcon, ColIcon, LocationIcon, RowIcon, TextIcon, ValueIcon };
