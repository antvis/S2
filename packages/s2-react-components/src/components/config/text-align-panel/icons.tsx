import React from 'react';

type RadioIconProps = {
  active: boolean;
};

export const LeftAlignIcon = React.memo<RadioIconProps>(({ active }) => {
  if (active) {
    return (
      <svg className="si" width="12px" height="10px" viewBox="0 0 12 10">
        <g
          id="设计输出"
          stroke="none"
          strokeWidth="1"
          fill="none"
          fillRule="evenodd"
        >
          <g
            transform="translate(-309.000000, -5352.000000)"
            fill="#3471F9"
            fillRule="nonzero"
            id="编组-31"
          >
            <g transform="translate(127.000000, 4744.000000)">
              <g id="编组-16" transform="translate(28.000000, 569.000000)">
                <g id="编组-15" transform="translate(152.000000, 36.000000)">
                  <g id="编组" transform="translate(2.000000, 3.000000)">
                    <path
                      d="M11.238,1.045 L0.76,1.045 C0.341,1.045 0,0.81 0,0.522 C0,0.234 0.341,0 0.76,0 L11.24,0 C11.659,0 12,0.234 12,0.522 C11.998,0.812 11.659,1.045 11.238,1.045 Z M7.024,4.03 L0.474,4.03 C0.214,4.03 0,3.795 0,3.507 C0,3.219 0.213,2.985 0.475,2.985 L7.025,2.985 C7.287,2.985 7.5,3.219 7.5,3.507 C7.499,3.795 7.287,4.03 7.024,4.03 L7.024,4.03 Z M7.024,7.015 L0.474,7.015 C0.214,7.015 0,6.781 0,6.493 C0,6.205 0.213,5.97 0.475,5.97 L7.025,5.97 C7.287,5.97 7.5,6.205 7.5,6.493 C7.499,6.782 7.287,7.015 7.024,7.015 L7.024,7.015 Z M11.238,10 L0.76,10 C0.341,10 0,9.766 0,9.478 C0,9.19 0.341,8.955 0.76,8.955 L11.24,8.955 C11.659,8.955 12,9.19 12,9.478 C11.998,9.767 11.659,10 11.238,10 Z"
                      id="形状"
                    />
                  </g>
                </g>
              </g>
            </g>
          </g>
        </g>
      </svg>
    );
  }

  return (
    <svg className="si" width="12px" height="10px" viewBox="0 0 12 10">
      <g
        id="设计输出"
        stroke="none"
        strokeWidth="1"
        fill="none"
        fillRule="evenodd"
        opacity="0.45"
      >
        <g
          transform="translate(-269.000000, -5288.000000)"
          fill="#000000"
          fillRule="nonzero"
          id="编组-31"
        >
          <g transform="translate(127.000000, 4744.000000)">
            <g id="编组-16备份" transform="translate(28.000000, 537.000000)">
              <g id="编组-14" transform="translate(112.000000, 4.000000)">
                <g id="编组" transform="translate(2.000000, 3.000000)">
                  <path
                    d="M11.238,1.045 L0.76,1.045 C0.341,1.045 0,0.81 0,0.522 C0,0.234 0.341,0 0.76,0 L11.24,0 C11.659,0 12,0.234 12,0.522 C11.998,0.812 11.659,1.045 11.238,1.045 Z M7.024,4.03 L0.474,4.03 C0.214,4.03 0,3.795 0,3.507 C0,3.219 0.213,2.985 0.475,2.985 L7.025,2.985 C7.287,2.985 7.5,3.219 7.5,3.507 C7.499,3.795 7.287,4.03 7.024,4.03 L7.024,4.03 Z M7.024,7.015 L0.474,7.015 C0.214,7.015 0,6.781 0,6.493 C0,6.205 0.213,5.97 0.475,5.97 L7.025,5.97 C7.287,5.97 7.5,6.205 7.5,6.493 C7.499,6.782 7.287,7.015 7.024,7.015 L7.024,7.015 Z M11.238,10 L0.76,10 C0.341,10 0,9.766 0,9.478 C0,9.19 0.341,8.955 0.76,8.955 L11.24,8.955 C11.659,8.955 12,9.19 12,9.478 C11.998,9.767 11.659,10 11.238,10 Z"
                    id="形状"
                  />
                </g>
              </g>
            </g>
          </g>
        </g>
      </g>
    </svg>
  );
});

export const CenterAlignIcon = React.memo<RadioIconProps>(({ active }) => {
  if (active) {
    return (
      <svg className="si" width="12px" height="11px" viewBox="0 0 12 11">
        <g
          id="设计输出"
          stroke="none"
          strokeWidth="1"
          fill="none"
          fillRule="evenodd"
        >
          <g
            transform="translate(-301.000000, -5288.000000)"
            fill="#3471F9"
            fillRule="nonzero"
            id="编组-31"
          >
            <g transform="translate(127.000000, 4744.000000)">
              <g id="编组-16备份" transform="translate(28.000000, 537.000000)">
                <g id="编组-14" transform="translate(112.000000, 4.000000)">
                  <g id="编组-33" transform="translate(32.000000, 0.000000)">
                    <g id="编组" transform="translate(2.000000, 3.000000)">
                      <path
                        d="M11.238,4.045 L0.76,4.045 C0.341,4.045 0,3.81 0,3.522 C0,3.234 0.341,3 0.76,3 L11.24,3 C11.659,3 12,3.234 12,3.522 C11.998,3.812 11.659,4.045 11.238,4.045 Z M9.024,1.045 L2.474,1.045 C2.214,1.045 2,0.81 2,0.522 C2,0.234 2.213,0 2.475,0 L9.025,0 C9.287,0 9.5,0.234 9.5,0.522 C9.499,0.81 9.287,1.045 9.024,1.045 L9.024,1.045 Z M9.024,7.045 L2.474,7.045 C2.214,7.045 2,6.81 2,6.522 C2,6.234 2.213,6 2.475,6 L9.025,6 C9.287,6 9.5,6.234 9.5,6.522 C9.499,6.812 9.287,7.045 9.024,7.045 L9.024,7.045 Z M11.238,10.045 L0.76,10.045 C0.341,10.045 0,9.81 0,9.522 C0,9.234 0.341,9 0.76,9 L11.24,9 C11.659,9 12,9.234 12,9.522 C11.998,9.812 11.659,10.045 11.238,10.045 Z"
                        id="形状"
                      />
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </g>
        </g>
      </svg>
    );
  }

  return (
    <svg className="si" width="12px" height="11px" viewBox="0 0 12 11">
      <g
        id="设计输出"
        stroke="none"
        strokeWidth="1"
        fill="none"
        fillRule="evenodd"
        opacity="0.45"
      >
        <g
          transform="translate(-341.000000, -5352.000000)"
          fill="#000000"
          fillRule="nonzero"
          id="编组-31"
        >
          <g transform="translate(127.000000, 4744.000000)">
            <g id="编组-16" transform="translate(28.000000, 569.000000)">
              <g id="编组-15" transform="translate(152.000000, 36.000000)">
                <g id="编组-33备份" transform="translate(32.000000, 0.000000)">
                  <g id="编组" transform="translate(2.000000, 3.000000)">
                    <path
                      d="M11.238,4.045 L0.76,4.045 C0.341,4.045 0,3.81 0,3.522 C0,3.234 0.341,3 0.76,3 L11.24,3 C11.659,3 12,3.234 12,3.522 C11.998,3.812 11.659,4.045 11.238,4.045 Z M9.024,1.045 L2.474,1.045 C2.214,1.045 2,0.81 2,0.522 C2,0.234 2.213,0 2.475,0 L9.025,0 C9.287,0 9.5,0.234 9.5,0.522 C9.499,0.81 9.287,1.045 9.024,1.045 L9.024,1.045 Z M9.024,7.045 L2.474,7.045 C2.214,7.045 2,6.81 2,6.522 C2,6.234 2.213,6 2.475,6 L9.025,6 C9.287,6 9.5,6.234 9.5,6.522 C9.499,6.812 9.287,7.045 9.024,7.045 L9.024,7.045 Z M11.238,10.045 L0.76,10.045 C0.341,10.045 0,9.81 0,9.522 C0,9.234 0.341,9 0.76,9 L11.24,9 C11.659,9 12,9.234 12,9.522 C11.998,9.812 11.659,10.045 11.238,10.045 Z"
                      id="形状"
                    />
                  </g>
                </g>
              </g>
            </g>
          </g>
        </g>
      </g>
    </svg>
  );
});

export const RightAlignIcon = React.memo<RadioIconProps>(({ active }) => {
  if (active) {
    return (
      <svg className="si" width="12px" height="10px" viewBox="0 0 12 10">
        <g
          id="设计输出"
          stroke="none"
          strokeWidth="1"
          fill="none"
          fillRule="evenodd"
          opacity="1"
        >
          <g
            transform="translate(-333.000000, -5288.000000)"
            fill="#3471F9"
            fillRule="nonzero"
            id="编组-31"
          >
            <g transform="translate(127.000000, 4744.000000)">
              <g id="编组-16备份" transform="translate(28.000000, 537.000000)">
                <g id="编组-14" transform="translate(112.000000, 4.000000)">
                  <g id="编组-32" transform="translate(64.000000, 0.000000)">
                    <g id="编组" transform="translate(2.000000, 3.000000)">
                      <path
                        d="M0.762,1.045 L11.24,1.045 C11.659,1.045 12,0.81 12,0.522 C12,0.234 11.659,0 11.24,0 L0.76,0 C0.34,0 0,0.234 0,0.522 C0.002,0.812 0.341,1.045 0.762,1.045 Z M4.976,4.03 L11.526,4.03 C11.786,4.03 12,3.795 12,3.507 C12,3.219 11.787,2.985 11.525,2.985 L4.975,2.985 C4.713,2.985 4.5,3.219 4.5,3.507 C4.501,3.795 4.713,4.03 4.976,4.03 Z M4.976,7.015 L11.526,7.015 C11.786,7.015 12,6.781 12,6.493 C12,6.205 11.787,5.97 11.525,5.97 L4.975,5.97 C4.713,5.97 4.5,6.205 4.5,6.493 C4.501,6.782 4.713,7.015 4.976,7.015 Z M0.762,10 L11.24,10 C11.659,10 12,9.766 12,9.478 C12,9.19 11.659,8.955 11.24,8.955 L0.76,8.955 C0.341,8.955 0,9.19 0,9.478 C0.002,9.767 0.341,10 0.762,10 Z"
                        id="形状"
                      />
                    </g>
                  </g>
                </g>
              </g>
            </g>
          </g>
        </g>
      </svg>
    );
  }

  return (
    <svg className="si" width="12px" height="10px" viewBox="0 0 12 10">
      <g
        id="设计输出"
        stroke="none"
        strokeWidth="1"
        fill="none"
        fillRule="evenodd"
        opacity="0.45"
      >
        <g
          transform="translate(-333.000000, -5288.000000)"
          fill="#000000"
          fillRule="nonzero"
          id="编组-31"
        >
          <g transform="translate(127.000000, 4744.000000)">
            <g id="编组-16备份" transform="translate(28.000000, 537.000000)">
              <g id="编组-14" transform="translate(112.000000, 4.000000)">
                <g id="编组-32" transform="translate(64.000000, 0.000000)">
                  <g id="编组" transform="translate(2.000000, 3.000000)">
                    <path
                      d="M0.762,1.045 L11.24,1.045 C11.659,1.045 12,0.81 12,0.522 C12,0.234 11.659,0 11.24,0 L0.76,0 C0.34,0 0,0.234 0,0.522 C0.002,0.812 0.341,1.045 0.762,1.045 Z M4.976,4.03 L11.526,4.03 C11.786,4.03 12,3.795 12,3.507 C12,3.219 11.787,2.985 11.525,2.985 L4.975,2.985 C4.713,2.985 4.5,3.219 4.5,3.507 C4.501,3.795 4.713,4.03 4.976,4.03 Z M4.976,7.015 L11.526,7.015 C11.786,7.015 12,6.781 12,6.493 C12,6.205 11.787,5.97 11.525,5.97 L4.975,5.97 C4.713,5.97 4.5,6.205 4.5,6.493 C4.501,6.782 4.713,7.015 4.976,7.015 Z M0.762,10 L11.24,10 C11.659,10 12,9.766 12,9.478 C12,9.19 11.659,8.955 11.24,8.955 L0.76,8.955 C0.341,8.955 0,9.19 0,9.478 C0.002,9.767 0.341,10 0.762,10 Z"
                      id="形状"
                    />
                  </g>
                </g>
              </g>
            </g>
          </g>
        </g>
      </g>
    </svg>
  );
});
