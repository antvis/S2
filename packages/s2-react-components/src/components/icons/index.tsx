import { ReactComponent as ColIcon } from '@antv/s2-shared/src/icons/col-icon.svg';
import { ReactComponent as RowIcon } from '@antv/s2-shared/src/icons/row-icon.svg';
import { ReactComponent as ValueIcon } from '@antv/s2-shared/src/icons/value-icon.svg';
import './index.less';

export { ColIcon, RowIcon, ValueIcon };
