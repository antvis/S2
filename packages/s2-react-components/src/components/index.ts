export * from './common';

export * from './advanced-sort';
export * from './drill-down';
export * from './export';
export * from './switcher';

export * from './frozen-panel';
export * from './text-align-panel';
export * from './theme-panel';
