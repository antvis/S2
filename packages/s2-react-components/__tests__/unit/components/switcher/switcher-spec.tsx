import { Switcher } from '@/components';
import { UpCircleOutlined } from '@ant-design/icons';
import { render, screen } from '@testing-library/react';
import React from 'react';

describe('switcher component test', () => {
  test('should render component', () => {
    const { asFragment, container } = render(<Switcher title="行列切换" />);

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('行列切换')).toBeDefined();
    expect(container.querySelector('.anticon-swap')).toBeDefined();
  });

  test('should render custom icon', () => {
    const { asFragment, container } = render(
      <Switcher icon={<UpCircleOutlined />} />,
    );

    expect(asFragment()).toMatchSnapshot();
    expect(container.querySelector('.anticon-up-circle')).toBeDefined();
  });

  test('should render custom children', () => {
    const { asFragment } = render(
      <Switcher>
        <div>自定义</div>
      </Switcher>,
    );

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('自定义')).toBeDefined();
  });

  test('should render custom popover props', () => {
    const { asFragment } = render(
      <Switcher
        popover={{
          open: true,
          className: 'custom-popover',
          content: <div>自定义内容</div>,
        }}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('自定义内容')).toBeDefined();
  });
});
