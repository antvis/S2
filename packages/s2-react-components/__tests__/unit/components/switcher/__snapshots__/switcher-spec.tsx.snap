// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`switcher component test should render component 1`] = `
<DocumentFragment>
  <button
    class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-default ant-btn-sm antv-s2-switcher-entry-button"
    type="button"
  >
    <span
      class="ant-btn-icon"
    >
      <span
        aria-label="swap"
        class="anticon anticon-swap"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="swap"
          fill="currentColor"
          focusable="false"
          height="1em"
          style="transform: rotate(90deg);"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M847.9 592H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h605.2L612.9 851c-4.1 5.2-.4 13 6.3 13h72.5c4.9 0 9.5-2.2 12.6-6.1l168.8-214.1c16.5-21 1.6-51.8-25.2-51.8zM872 356H266.8l144.3-183c4.1-5.2.4-13-6.3-13h-72.5c-4.9 0-9.5 2.2-12.6 6.1L150.9 380.2c-16.5 21-1.6 51.8 25.1 51.8h696c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </span>
    </span>
    <span>
      行列切换
    </span>
  </button>
</DocumentFragment>
`;

exports[`switcher component test should render custom children 1`] = `
<DocumentFragment>
  <div>
    自定义
  </div>
</DocumentFragment>
`;

exports[`switcher component test should render custom icon 1`] = `
<DocumentFragment>
  <button
    class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-default ant-btn-sm antv-s2-switcher-entry-button"
    type="button"
  >
    <span
      class="ant-btn-icon"
    >
      <span
        aria-label="up-circle"
        class="anticon anticon-up-circle"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="up-circle"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M518.5 360.3a7.95 7.95 0 00-12.9 0l-178 246c-3.8 5.3 0 12.7 6.5 12.7H381c10.2 0 19.9-4.9 25.9-13.2L512 460.4l105.2 145.4c6 8.3 15.6 13.2 25.9 13.2H690c6.5 0 10.3-7.4 6.5-12.7l-178-246z"
          />
          <path
            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
          />
        </svg>
      </span>
    </span>
    <span>
      行列切换
    </span>
  </button>
</DocumentFragment>
`;

exports[`switcher component test should render custom popover props 1`] = `
<DocumentFragment>
  <button
    class="ant-btn css-dev-only-do-not-override-3rel02 ant-btn-default ant-btn-sm antv-s2-switcher-entry-button ant-popover-open custom-popover"
    type="button"
  >
    <span
      class="ant-btn-icon"
    >
      <span
        aria-label="swap"
        class="anticon anticon-swap"
        role="img"
      >
        <svg
          aria-hidden="true"
          data-icon="swap"
          fill="currentColor"
          focusable="false"
          height="1em"
          style="transform: rotate(90deg);"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M847.9 592H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h605.2L612.9 851c-4.1 5.2-.4 13 6.3 13h72.5c4.9 0 9.5-2.2 12.6-6.1l168.8-214.1c16.5-21 1.6-51.8-25.2-51.8zM872 356H266.8l144.3-183c4.1-5.2.4-13-6.3-13h-72.5c-4.9 0-9.5 2.2-12.6 6.1L150.9 380.2c-16.5 21-1.6 51.8 25.1 51.8h696c4.4 0 8-3.6 8-8v-60c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </span>
    </span>
    <span>
      行列切换
    </span>
  </button>
</DocumentFragment>
`;
